modules/react/button/lib/PrimaryButtonKing.tsx

```diff
@@ -0,0 +1,423 @@
+  children?: React.ReactNode;
+}
+
+// Button sizes where data labels are enabled
+const dataLabelSizes = ['medium', 'large'];
+
+const StyledPrimaryButton = styled(Button)<StyledType & PrimaryButtonKingProps>(
+  {
+    ...type.levels.subtext.large,
+    cursor: 'pointer',
+    display: 'inline-flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    boxSizing: 'border-box',
+    boxShadow: 'none',
+    outline: 'none',
+    fontWeight: type.properties.fontWeights.bold,
+    lineHeight: 'normal',
+    whiteSpace: 'nowrap',
+    WebkitFontSmoothing: 'antialiased',
+    MozOsxFontSmoothing: 'grayscale',
+    borderRadius: borderRadius.circle,
+    position: 'relative',
+    verticalAlign: 'middle',
+    overflow: 'hidden',
+    border: '1px solid transparent',
+    background: 'red',
+    transition:
+      'box-shadow 120ms linear, border 120ms linear, background-color 120ms linear, color 120ms linear',
+    '&:disabled, &:disabled:active': {
+      cursor: 'default',
+      boxShadow: 'none',
+      opacity: 0.4,
+    },
+
+    '&:hover:active': {transitionDuration: '40ms'},
+
+    '& > *:first-of-type': {
+      marginLeft: '0',
+    },
+
+    '& > *:last-of-type': {
+      marginRight: 0,
+    },
+  },
+  ({size}) => {
+    switch (size) {
+      case 'large':
+        return {
+          ...type.levels.body.small,
+          fontWeight: type.properties.fontWeights.bold,
+          height: '48px',
+          '& > * ': {
+            margin: `0 ${spaceNumbers.xxs / 2}px`,
+          },
+        };
+      case 'medium':
+      default:
+        return {
+          height: space.xl,
+          '& > * ': {
+            margin: `0 ${spaceNumbers.xxs / 2}px`,
+          },
+        };
+      case 'small':
+        return {
+          height: space.l,
+          '& > * ': {
+            margin: `0 ${spaceNumbers.xxxs / 2}px`,
+          },
+        };
+      case 'extraSmall':
+        return {
+          ...type.levels.subtext.medium,
+          fontWeight: type.properties.fontWeights.bold,
+          height: space.m,
+          '& > * ': {
+            margin: `0 ${spaceNumbers.xxxs / 2}px`,
+          },
+        };
+    }
+  },
+  ({grow}) => grow && {width: '100%', maxWidth: '100%'},
+  ({variant, theme}) => {
+    if (!getPrimaryButtonColorsNew(variant, theme)) {
+      return;
+    }
+
+    const colors: ButtonColors = getPrimaryButtonColorsNew(variant, theme);
+    const baseStyles = {
+      backgroundColor: colors.default.background,
+      borderColor: colors.default.border,
+      color: colors.default.label,
+      ...(colors.default.icon && {
+        '.wd-icon-fill, .wd-icon-accent, .wd-icon-accent2, .wd-icon-background': {
+          transition: 'fill 120ms ease-in',
+        },
+        ...getIconColorSelectorsNew(theme, colors.default.icon, false),
+      }),
+      ...(colors.default.labelData && {
+        ['.' + buttonLabelDataClassName]: {
+          color: colors.default.labelData,
+        },
+      }),
+    };
```

Comment:
I wonder if the styles will be the same between `Primary`, `Secondary`, and `Tertiary` buttons. If yes, I think this is a good candidate for a shared function. If not, this could be the way to go. Moving it out to another function increases the abstraction level and makes it harder to understand and change, but if our goal is to simplify buttons by making the layout the same and only colors change, an imported function helps to support that goal.